### PR TITLE
Stairs: fix rotation on place

### PR DESCRIPTION
**Pull request by alek13**
For example. slabs should placed vertically when places on a "wall".    
    
    use builtin \`rotate_node()\`:
    
    - so, use internal \`core.is_creative_enabled\`
    - fix rotation on stairs & slabs place
    - fix double \"take_item\" in creative mode

### CODE REVIEW

File: mods/stairs/init.lua [hunk 1]
```diff
@@ -194,36 +168,11 @@ function stairs.register_slab(subname, recipeitem, groups, images, description,
...
-					itemstack:take_item()
-				end
-				return itemstack
-			else
-				return rotate_and_place(itemstack, placer, pointed_thing)
+			if pointed_thing.type ~= "node" then
+				return itemstack, nil
 			end
+
+			return minminetest.rotate_node(itemstack, placer, pointed_thing)
```
Comment (MoNTE48) [inline mods/stairs/init.lua:175]: O_o

Comment (alek13) [reply mods/stairs/init.lua]: oops :)  fixed

**Comment by sfan5** [comment]
PR must merge into `master`, not `stable-5`.

**Comment by alek13** [comment]
@sfan5 
If I understand correctly, `master` branch for now have changes for MT `5.4.0-dev` and possible uses new functionality, which it does not compatible with `5.3` (in general).

So if I want to fix this for `5.3` version, I need to PR into some corresponding branch (usually with same name - `5.3`).

Is there some way to fix this for `5.3` or I misunderstand the workflow ?

**Comment by sfan5** [comment]
Any new changes are put into `master`, then when 5.4.0 is released the current state of `master` is pushed into `stable-5`.
The only exception to this are backport releases, which we don't do currently (there will be no 5.3.1).

You cannot retroactively fix this in 5.3, but what you *can* do is to fix the bug for the next version.

**Comment by alek13** [comment]
@sfan5 
Sadly... :)
Maybe it would be better to create corresponding branches and when it have updates re-release MT with new MTG with patch up (`5.3.1`) ? Sounds crazy? :)
Or maybe some else solution ?

**Comment by alek13** [comment]
@sfan5 @rubenwardy 
Maybe best of all
 - create corresponding branches (`5.1`, `5.2`, `5.3`)
 - release MT without MTG and on first launch ask to download it
 - and than everyone can update MTG if there is updates for theirs version (in corresponding branch)
?

**Comment by alek13** [comment]
rebased on `master`

**Comment by alek13** [comment]
> @sfan5
> Sadly... :)
> Maybe it would be better to create corresponding branches and when it have updates re-release MT with new MTG with patch up (5.3.1) ? Sounds crazy? :)
> Or maybe some else solution ?

> @sfan5 @rubenwardy
> Maybe best of all
> - create corresponding branches (`5.1`, `5.2`, `5.3`)
> - release MT without MTG and on first launch ask to download it
> - and than everyone can update MTG if there is updates for theirs version (in corresponding branch)
> 
> ?

@sfan5 , @rubenwardy 
any thoughts ?
does it make sense to put it in a separate thread (issue) for discussion with others ?

**Comment by alek13** [comment]
@sfan5 can we merge this ?

**Comment by sfan5** [comment]
I haven't got around to reviewing and testing this PR yet, after I (or someone else) has this can be merged.

Regarding review: I'm wondering what happens to the special handling for `stairs:slab_`, does that just disappear?

**Comment by alek13** [comment]
not sure about what "special handling" you are meen, but all functionality works as before

**Review by SmallJoker — CHANGES_REQUESTED** [review]
Side-place on stair

This code is (at least partially) required to allow placing same stairs in a column (place to its side face). This PR breaks a very convenient way of building rooftops because the stairs are now tilted as shown above.

**Comment by sfan5** [comment]
Stale, closing.

**Pull request closed without merging**